cla_backend/apps/reports/sql/CaseDemographicsReport.sql

```diff
@@ -0,0 +1,151 @@
+  END as "Adjustments_LLI"
+  ,COALESCE(adapt.minicom, false)::bool as "Adjustments_MIN"
+  ,COALESCE(adapt.text_relay, false)::bool as "Adjustments_TYP"
+  ,COALESCE(adapt.callback_preference, false)::bool as "Adjustments_CallbackPreferred"
+  ,COALESCE(adapt.skype_webcam, false)::bool as "Adjustments_Skype"
+  ,CASE
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) <= 24 THEN 'A'
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) <= 29 THEN 'B'
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) <= 39 THEN 'C'
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) <= 49 THEN 'D'
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) <= 59 THEN 'E'
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) <= 79 THEN 'F'
+      WHEN EXTRACT(YEAR from age(now(), pd.date_of_birth)) >= 80 THEN 'G'
+      WHEN pd.date_of_birth IS NULL THEN 'U'
+      END as "Age(Range)"
+  ,mc.code as "Media_Code"
+  ,c.source as "Contact_Type"
+  ,null as "Referral_Agencies"
+  ,c.exempt_user_reason as "Exempt_Client"
+  ,CASE upper(adapt.language) 
+    WHEN upper('Welsh') THEN true 
+    ELSE false 
+  END as "Welsh"
+  ,adapt.language as "Language"
+  ,log.created as "Outcome_Created_At"
+  ,c.thirdparty_details_id::bool as "Has_Third_Party"
+  ,cc_org.name as "Organisation"
+  ,c.notes as "Notes"
+  ,c.provider_notes as "Provider Notes"
+  ,adapt.notes as "Adaptation Notes"
+  ,pd.vulnerable_user as "Vulnerable User"
+  ,CASE 
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('DE', 'LE', 'LN', 'NG') THEN 'East Midlands'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('AL', 'CB', 'CM', 'CO', 'HP', 'IP', 'LU', 'NR', 'SG', 'SS') THEN 'East of England'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('BR', 'CR', 'DA', 'E', 'EC', 'EN', 'HA', 'IG', 'KT', 'N', 'NW', 'RM', 'SE', 'SM', 'SW', 'TW', 'UB', 'W', 'WC', 'WD') THEN 'Greater London'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('DH', 'DL', 'NE', 'SR', 'TS') THEN 'North East'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('BB', 'BD', 'BL', 'CA', 'CH', 'CW', 'FY', 'HD', 'L', 'LA', 'M', 'OL', 'PR', 'SK', 'WA', 'WN') THEN 'North West'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('DN', 'HD', 'HG', 'HU', 'HX', 'LS', 'S', 'WF', 'YO') THEN 'Yorkshire and Humber'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('BN', 'CT', 'GU', 'ME', 'MK', 'OX', 'PO', 'RG', 'RH', 'SL', 'SO', 'TN') THEN 'South East'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('BA', 'BH', 'BS', 'DT', 'EX', 'GL', 'PL', 'SN', 'SP', 'TA', 'TQ', 'TR') THEN 'South West'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('CF', 'LD', 'LL', 'NP', 'SA', 'SY') THEN 'Wales'
+    WHEN TRIM('123456789' FROM SUBSTRING(pd.postcode, 1, 2)) IN ('B', 'CV', 'DY', 'HR', 'NN', 'ST', 'TF', 'WR', 'WS', 'WV') THEN 'West Midlands'
+    ELSE 'NA'
+  END as "Geographical_region"
```

Comment:
The ticket makes reference to [this list of the regions of England](https://en.wikipedia.org/wiki/Regions_of_England#List_of_regions), which lists 'Yorkshire and the Humber' as a separate region to the 'North East', however, this region is merged with the 'North East' in your current version. This could cause issues if, for example, the population of the 'North East' region was used in a later calculation and this encompassed a larger region than expected. I tried to find a more official source for the regions of England and the best I could find was the [ONS's analysis of the 2021 Census](https://www.ons.gov.uk/economy/grossdomesticproductgdp/bulletins/gdpukregionsandcountries/julytoseptember2022#:~:text=5.,UK%20regions%20and%20countries%20data&text=Quarterly%20economic%20activity%20within%20Wales,East%2C%20and%20South%20West).), which also lists 'Yorkshire and The Humber' as a separate region.

Reply:
Using the data from this link https://postcodebyaddress.co.uk/region/yorkshire-and-the-humber I amended the SQL to take into account Yorkshire and Humber.